### PR TITLE
add kubernetes-secrets-injector chart

### PR DESCRIPTION
**Pull request by volodymyrZotov**

### CODE REVIEW

File: ct.yaml [hunk 1]
```diff
@@ -5,3 +5,4 @@ debug: true
 helm-extra-args: --timeout 120s
 charts:
   - charts/connect
+  - charts/kubernetes-secrets-injector
```
Comment (edif2008) [inline ct.yaml:8]: Will it be ok if we only say `secrets-injector`? Helm is already used for Kubernetes, so saying it again here may not be the most useful.

Comment (volodymyrZotov) [reply ct.yaml:8]: That makes sense.
Then maybe also rename the folder also? `charts/kubernetes-secrets-injector` to `charts/secrets-injector`?

File: charts/secrets-injector/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+or 
+
+```bash
+$ helm install --set injector.applicationName=injector injector ./secrets-injector
+```
+
+### Values
+| Key | Type | Default | Description |
+|-----|------|---------|-------------|
+| injector.applicationName | string | `"secrets-injector"` | The name of 1Password Kubernetes Secrets Injector Application |
```
Comment (jillianwilson) [inline charts/secrets-injector/README.md:34]: should we add a value for configuring the version of the injector here as well in case you don't want the latest?

File: charts/secrets-injector/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      app: {{ .Values.injector.applicationName }}
+  template:
+    metadata:
+      labels:
+        app: {{ .Values.injector.applicationName }}
+    spec:
+      serviceAccountName: {{ .Values.injector.applicationName }}
+      containers:
+        - name: {{ .Values.injector.applicationName }}
+          image: {{ .Values.injector.imageRepository }}:{{ .Chart.AppVersion }}
```
Comment (edif2008) [inline charts/secrets-injector/templates/deployment.yaml:24]: This gives me the impression that the user can't change the version of the image. We should enable the users to do that.

I see that for Connect we have something like this:
```yaml
image: {{ .Values.connect.api.imageRepository }}:{{ tpl .Values.connect.version . }}
```

Maybe something similar works here as well?

File: charts/secrets-injector/README.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# 1Password Kubernetes Secrets Injector Helm chart
+Installing the Helm Chart with default configurations will deploy 1Password Kubernetes Secrets Injector in your default Namespace.
+
+## Installation
+```
+helm install --generate-name 1password/secrets-injector
+```
+
+## Usage
+Please follow [Secrets Injector documentation](https://github.com/1Password/kubernetes-secrets-injector#1password-secrets-injector-for-kubernetes)
```
Comment (edif2008) [inline charts/secrets-injector/README.md:10]: This feel a bit too small to include it as a section.

We have 2 options here:
- Remove it altogether (I see we don't include this for Connect Helm Charts)
- Rephrase it to make it fit better. Maybe something like this:
  ```suggestion
  ## Usage
  Check out the [documentation for the Secrets Injector](https://github.com/1Password/kubernetes-secrets-injector#1password-secrets-injector-for-kubernetes) to see how to use it in your infrastructure.
  ```

File: charts/secrets-injector/README.md (hunk 2 above)
Comment (volodymyrZotov) [reply charts/secrets-injector/README.md:34]: It uses the chart AppVersion
```
image: {{ .Values.injector.imageRepository }}:{{ .Chart.AppVersion }}
```
So, we always use the explicit version.

File: charts/secrets-injector/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+          - 2>&1
+          env:
+          - name: POD_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+          lifecycle:
+            preStop:
+              exec:
+                command: [ "/bin/sh", "-c", "/prestop.sh" ]
```
[the remark this answers is not in the document]
Comment (volodymyrZotov) [reply charts/secrets-injector/templates/deployment.yaml:39]: I think we do not need it, as we copy it while building the injector image
```
# from injector Dockerfile
# install the prestop script
COPY ./prestop.sh .
```

File: charts/secrets-injector/README.md (hunk 4 above)
Comment (volodymyrZotov) [reply charts/secrets-injector/README.md:10]: I removed it

File: charts/secrets-injector/README.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
+# 1Password Kubernetes Secrets Injector Helm chart
+Installing the Helm Chart with default configurations will deploy 1Password Kubernetes Secrets Injector in your default Namespace.
```
[the remark this answers is not in the document]
Comment (volodymyrZotov) [reply charts/secrets-injector/README.md:2]: moved it

File: charts/secrets-injector/templates/deployment.yaml (hunk 3 above)
Comment (volodymyrZotov) [reply charts/secrets-injector/templates/deployment.yaml:24]: Yep, that works! Updated.

File: charts/secrets-injector/templates/deployment.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (edif2008) [reply charts/secrets-injector/templates/deployment.yaml:39]: Awesome. Thanks for double-checking this. 😄

File: charts/secrets-injector/templates/deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+metadata:
+  name: {{ .Values.injector.applicationName }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ .Values.injector.applicationName }}
+  annotations:
+    helm.sh/hook: pre-install
+    helm.sh/hook-weight: "1"
+spec:
+  replicas: {{ .Values.injector.replicas }}
```
[the remark this answers is not in the document]
Comment (edif2008) [reply charts/secrets-injector/templates/deployment.yaml:12]: Sounds good to me, then.

[the remark this answers is not in the document]
Comment (edif2008) [reply charts/secrets-injector/templates/deployment.yaml:12]: I’m still not convinced if it makes sense to have replicas for the webhook.
Usually you need only one webhook in your infra….so why have multiple pods that handle the same webhook?
I may have another perspective here, but I’m glad to hear your perspective.

Adding it just for the sake of it (or cause user asked for it in other places) doesn’t feel like a good enough reason.
For Connect it makes sense to have multiple instances of it in your infra, thus multiple replicas…
Not sure if it’s the same for the webhook.

[the remark this answers is not in the document]
Comment (volodymyrZotov) [reply charts/secrets-injector/templates/deployment.yaml]: As per our discussion, we just remove it.

File: charts/secrets-injector/README.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+### Values
+| Key | Type | Default | Description |
+|-----|------|---------|-------------|
+| injector.applicationName | string | `"secrets-injector"` | The name of 1Password Kubernetes Secrets Injector Application |
+| injector.port | string | `443` | The port the Secrets Injector exposes |
+| injector.targetPort | integer | `8443` | The port the Secrets Injector API sends requests to the pod |
+| injector.imageRepository | string | `"1password/kubernetes-secrets-injector"` | The 1Password Secrets Injector docker image repository |
+| injector.imagePullPolicy | string | `"IfNotPresent"` | The 1Password Secrets Injector docker image policy. `"IfNotPresent"` means the image is pulled only if it is not already present locally. |
+| injector.version         | string | `{{.Chart.AppVersion}}` | The 1Password Secrets Injector version to pull. |
+| injector.replicas        | integer | `1` | Amount of the Secrets Injector replicas. |
```
[the remark this answers is not in the document]
Comment (volodymyrZotov) [reply charts/secrets-injector/README.md:36]: Changed to alphabetical order

**Review by jillianwilson — APPROVED** [review]
